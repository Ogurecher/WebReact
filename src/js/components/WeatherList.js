import React, { Component } from "react";
import WeatherListItem from "./WeatherListItem.js"

export default class WeatherList extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <ul>
                <li><WeatherListItem items={[this.props.weather.weather[0].main, `; ${this.props.weather.main.temp}C`]} /></li>
                <li><WeatherListItem items={[`Wind direction: ${this.props.weather.wind.deg}deg`, `; Wind speed: ${this.props.weather.wind.speed} m/s`]} /></li>
                <li><WeatherListItem items={[`Pressure: ${this.props.weather.main.pressure} hpa`]} /></li>
                <li><WeatherListItem items={[`Humidity: ${this.props.weather.main.humidity}% `]} /></li>
                <li><WeatherListItem items={[`Latitude: ${this.props.weather.coord.lat}`, `; Longitude: ${this.props.weather.coord.lon}`]} /></li>
            </ul>
        );
    }
}