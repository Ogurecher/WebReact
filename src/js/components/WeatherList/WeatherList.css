.city-info-weather-list {
    flex: 1;
    flex-basis: 100%;
}