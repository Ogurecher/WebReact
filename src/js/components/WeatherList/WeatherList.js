import React, { Component } from 'react';
import WeatherListItem from '../WeatherListItem/WeatherListItem';
import './WeatherList.css';

export default class WeatherList extends Component {

    render() {
        return (
            <div className='city-info-weather-list'>
                <WeatherListItem label='Wind' content={[`${this.props.weather.wind.deg}deg`, `; ${this.props.weather.wind.speed} m/s`]} />
                <WeatherListItem label='Clouds'  content={[this.props.weather.weather[0].description]}/>
                <WeatherListItem label='Pressure' content={[`${this.props.weather.main.pressure} hpa`]} />
                <WeatherListItem label='Humidity' content={[`${this.props.weather.main.humidity}%`]} />
                <WeatherListItem label='Coordinates' content={[`[${this.props.weather.coord.lat}`, `; ${this.props.weather.coord.lon}]`]} />
            </div>
        );
    }
}