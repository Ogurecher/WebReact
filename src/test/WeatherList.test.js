import React from 'react';
import renderer from 'react-test-renderer';

import WeatherList from '../js/components/WeatherList/WeatherList';

describe('WeatherList', function() {
    test('snapshot renders', function() {
        const weatherStub = {
            wind: {
                deg: 'winddeg',
                speed: 'windspd'
            },
            weather: [{
                description: 'desc'
            }],
            main: {
                pressure: 'pressure',
                humidity: 'humidity'
            },
            coord: {
                lat: 'lat',
                lon: 'lon'
            }
        };

        const component = renderer.create(<WeatherList weather={weatherStub} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});