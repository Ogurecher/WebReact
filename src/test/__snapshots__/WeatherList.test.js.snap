// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherList snapshot renders 1`] = `
<div
  className="city-info-weather-list"
>
  <div
    className="city-info-weather-list-item"
  >
    <p>
       
      Wind
       
    </p>
    <p>
       
      winddegdeg
      ; windspd m/s
       
    </p>
  </div>
  <div
    className="city-info-weather-list-item"
  >
    <p>
       
      Clouds
       
    </p>
    <p>
       
      desc
       
    </p>
  </div>
  <div
    className="city-info-weather-list-item"
  >
    <p>
       
      Pressure
       
    </p>
    <p>
       
      pressure hpa
       
    </p>
  </div>
  <div
    className="city-info-weather-list-item"
  >
    <p>
       
      Humidity
       
    </p>
    <p>
       
      humidity%
       
    </p>
  </div>
  <div
    className="city-info-weather-list-item"
  >
    <p>
       
      Coordinates
       
    </p>
    <p>
       
      [lat
      ; lon]
       
    </p>
  </div>
</div>
`;
